contrib/http_jetty_client/src/main/java/io/opencensus/contrib/http/jetty/client/HttpRequestListener.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http.jetty.client;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.contrib.http.HttpClientHandler;
+import io.opencensus.trace.Span;
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+
+/** This class extracts attributes from Http Client Request and Response. */
+@ExperimentalApi
+public final class HttpRequestListener implements Request.Listener, Response.CompleteListener {
+
+  private final Span parent;
+  private final HttpClientHandler<Request, Response, Request> handler;
+  @Nullable @VisibleForTesting Span span = null;
+
+  HttpRequestListener(Span parent, HttpClientHandler<Request, Response, Request> handler) {
+    this.parent = parent;
+    this.handler = handler;
+  }
+
+  @Override
+  public void onComplete(Result result) {
+    if (span != null) {
+      handler.handleEnd(span, result.getResponse(), result.getFailure());
+    }
+  }
+
+  @Override
+  public void onBegin(Request request) {
+    span = handler.handleStart(parent, request, request);
+  }
+
+  @Override
+  public void onCommit(Request request) {}
+
+  @Override
+  public void onContent(Request request, ByteBuffer content) {}
```

Comment:
Consider if we should save the content size and maybe record the message event onCommit.

Reply:
I can get the content length from request, so no need to save. I'll add the message event on commit.